pkg/controller/function/function_controller.go

```diff
@@ -148,49 +288,251 @@ func (r *ReconcileFunction) Reconcile(request reconcile.Request) (reconcile.Resu
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		return nil
+
+	} else if err != nil {
+		log.Error(err, "Error while trying to create Knative Build", "namespace", deployBuild.Namespace, "name", deployBuild.Name)
+		return err
+	}
+
+	if !reflect.DeepEqual(deployBuild.Spec, foundBuild.Spec) && !compareBuildImages(foundBuild, imageName) {
+		// TODO Investigate
+		// Build is not being updated. Changing the status and the spec is not triggering the creation of a new Build Pod.
+		// The Build Pod is in state completed once the image is pushed. This state won't change a new Pod needs to be created.
+		// It should be created once the status of the Build is updated to Status{}.
+		// The Build Controller looks for Status.Cluster.PodName and if it is empty the creates a new Pod.
+		// The code bellow delete the old Pod and update the Build object with the new spec and status empty.
+		// However this is not working as expected.
+
+		// TODO - WORKAROUND: The function bellow delete, create and get a new Build object with the new spec. It is not the desire solution.
+		err := r.updateBuildFunctionImage(foundBuild, deployBuild, fn)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func compareBuildImages(foundBuild *buildv1alpha1.Build, imageName string) bool {
+	if foundBuild.Spec.Template != nil && len(foundBuild.Spec.Template.Arguments) > 0 {
+		args := foundBuild.Spec.Template.Arguments
+		for _, arg := range args {
+			if arg.Name == "IMAGE" && arg.Value == imageName {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// Update Build object
+func (r *ReconcileFunction) updateBuildFunctionImage(foundBuild *buildv1alpha1.Build, deployBuild *buildv1alpha1.Build, fn *runtimev1alpha1.Function) error {
+
+	err := r.updateFunctionStatus(fn, runtimev1alpha1.FunctionConditionUpdating)
 	if err != nil {
-		log.Error(err, "namespace", deployCm.Namespace, "name", deployCm.Name)
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	// Delete already existing Build with the old image-
+	// foundBuild = foundBuild.DeepCopy()
+	log.Info("Deleting Knative Build", "namespace", foundBuild.Namespace, "name", foundBuild.Name)
+	err = r.Delete(context.TODO(), foundBuild)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// create new Build with the new updated image
+	log.Info("Creating Knative Build", "namespace", deployBuild.Namespace, "name", deployBuild.Name)
+	err = r.Create(context.TODO(), deployBuild)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
 	}
-	hash := sha256.New()
-	hash.Write([]byte(foundCm.Data["handler.js"] + foundCm.Data["package.json"]))
-	functionSha := fmt.Sprintf("%x", hash.Sum(nil))
 
-	dockerRegistry := rnInfo.RegistryInfo
-	imageName := fmt.Sprintf("%s/%s-%s:%s", dockerRegistry, fn.Namespace, fn.Name, functionSha)
+	// Check if the Build instance can be get it.
```

Comment:
? i'm sorry, but what should this mean?

Reply:
It is just a check after create the Build, to see if can be retrieve it without error. Probably is not needed. Sould be removed?